app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -1,94 +1,77 @@
 module ManageIQ::Providers
   module Hawkular
-    class MiddlewareManager::RefreshParser
+    class Inventory::Parser::MiddlewareManager < ManagerRefresh::Inventory::Parser
       include ::Hawkular::ClientUtils
 
-      def self.ems_inv_to_hashes(ems, options = nil)
-        new(ems, options).ems_inv_to_hashes
-      end
-
-      def initialize(ems, _options = nil)
-        @ems = ems
-        @eaps = []
-        @data = {}
+      def initialize
         @data_index = {}
       end
 
-      def ems_inv_to_hashes
-        # the order of the method calls is important here, because they make use of @eaps and @data_index
+      def parse
+        # the order of the method calls is important here, because they make use of @data_index
```

Comment:
@israel-hdez I guess you are talking about `server_group`. Are you saying the `name` of the server group is not always unique? If so, storing a lookup by name might overwrite previously stored groups.
But I guess its a defficiancy in the hawkular api that is being worked on from the hawkular side.

So, ok for doing it in a follow up